dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java

```diff
@@ -67,17 +70,122 @@ public class ContainerFactoryImpl implements ContainerFactory {
     private final FileAssetAPI   fileAssetAPI   = APILocator.getFileAssetAPI();
     private final HostAPI        hostAPI        = APILocator.getHostAPI();
 	private final IdentifierAPI  identifierAPI  = APILocator.getIdentifierAPI();
+	private static ContainerSQL containerSQL = ContainerSQL.getInstance();
 
 	@Override
+	public Container find(final String inode) throws DotStateException, DotDataException {
+
+		Container container = containerCache.get(inode);
+
+		if(container==null){
+			final List<Map<String, Object>> containerResults = new DotConnect()
+					.setSQL(containerSQL.FIND_BY_INODE)
+					.addParam(inode)
+					.loadObjectResults();
+			if (containerResults.isEmpty()) {
+				Logger.debug(this, "Container with inode: " + inode + " not found");
+				return null;
+			}
+
+			container = (Container) TransformerLocator.createContainerTransformer(containerResults).findFirst();
+
+			if(container != null && container.getInode() != null) {
+				containerCache.add(inode, container);
+			}
+		}
+
+		return container;
+	}
+
 	public void save(final Container container) throws DotDataException {
-		HibernateUtil.save(container);
-		containerCache.remove(container);
+		save(container, UUIDGenerator.generateUuid());
 	}
 
-	@Override
-	public void save(final Container container, final String existingId) throws DotDataException {
-		HibernateUtil.saveWithPrimaryKey(container, existingId);
-		containerCache.remove(container);
+	@WrapInTransaction
+	public void save(final Container container, final String inode) throws DotDataException {

```

Comment:
This logic seems wrong.  If you pass in an inode, it should be used in the container.  The `inode` param is never used.

Reply:
yeah you are right. It should be removed or used. Let me address this. Thanks for your feedback @wezell

Reply:
This feedback has been addressed